src/main/java/javax/validation/TypeParameter.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Bean Validation API
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package javax.validation;
+
+import java.io.Serializable;
+import java.lang.reflect.TypeVariable;
+
+/**
+ * Serializable structure representing a type parameter in a generic declaration.
+ * <p>
+ * This is a serializable and very simplified replacement for {@link TypeVariable}.
+ * <p>
+ * For now, it only supports class level generic declaration.
+ *
+ * @author Guillaume Smet
+ * @since 2.0
+ */
+public class TypeParameter implements Serializable {
```

Comment:
I thought it might be sufficiently basic/non implementation specific. But you're right, I changed it to an interface.